### ISSUE TITLE
Unclear Where Data is Being Stored/Written

### ISSUE THREAD

**Issue by taylorbreannaray** [KonradIT/parler-py-api#5]
Hi there @KonradIT, I was testing out your unofficial API in an attempt to do some research with Parler. I was trying to run **02_multiple_hashtags.py**, only to find that it took an incredible amount of time to run—eventually I had to just stop it altogether. I figured this isn't _that_ absurd since I was trying to retrieve data for a large number of hashtags. So, with that thought, I just tried out one. I'm not sure how long this will take, but I do have an outstanding question about how all of this is working:

Where are these JSON files being written to in the first place? I have no way of looking at what's currently being collected because I don't have an understanding of that. Any help or clarification is appreciated. Thanks!

**Comment by KonradIT** [comment]
Taylor, on that experiment the data is stored in `data/hashtags/*`.

https://github.com/KonradIT/parler-py-api/blob/master/experiments/02_multiple_hashtags.py#L50

**Comment by taylorbreannaray** [comment]
Apologies, let me rephrase. I did note that place in the code from the beginning. However, I guess a better question would be, where is that directory located? There is no data or hashtags folders by default. Is this being stored in a DB somewhere?

**Comment by taylorbreannaray** [comment]
Also, while I've got your attention, how long should I expect that script to run for if I'm pulling data from ~20 hashtags. This would be a considerable amount of time, yes?

**Comment by KonradIT** [comment]
You need to create a folder called hashtags inside `data` first. The data directory is in the repo. You should run the script from `.` (`python experiments/02_multiple_hashtags.py`).

It does take a considerable amount of time.

**Comment by ExtraLime** [comment]
Hey taylor, like Konrad said you need to create your own storage. Its pretty easy to develop your own function for exactly what you need if you take it from the cli and into a jupyter notebook. Either way, with the response handler i just pull requested, i was able to scrape for 15 hours before I was disconnected.

Also it depends what you want from each hashtag. The comments take the most time and are the most problematic.

**Comment by taylorbreannaray** [comment]
Thank you @KonradIT and @ExtraLime for all of the helpful tips and clarification! I think I figured out what I need to do now. Naturally, I tried to make the process more complicated than it actually was.

Could the **02_multiple_hashtags.py** file be modified so that posts containing certain keywords _and_ certain hashtags could be pulled? For example, say I want posts related to COVID-19 that mention the term, but do not include any hashtag such as "covid_19", "covid19" or some other hashtag variant. Would this be simple to do?

**Comment by ExtraLime** [comment]
In my opinion, it would not be simple to do. The easiest route for you is to just collect as much 'COVID-19' as you can and then filter by your parameters afterwords. If my objective was to collect only posts that A. mentioned "COVID-19" AND  B. did not include hashtags(#X#Y#Z...), I would use the methods of the init.py and write a loop that calls the feed and filters the posts based on your parameters. I don't know how comfortable you are with python or bash to really give you a better answer.
